safespace/src/components/SendSMS.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react'
+
+export default class SendSMS extends Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            time : '',
+            phone : ''
+        }
+    }
+
+    handleChange = (e) => {
+        this.setState({
+            [e.target.name] : e.target.value
+        })
+    }
+
+      duration = (e,time) => {
+        const currentHours = new Date().getHours();
+        const currentMinutes = new Date().getMinutes();
+        const pieces = time.split(':');
+        const targetHours = parseInt(pieces[0],10);
+        const targetMinutes = parseInt(pieces[1],10);
+        const durationHours = targetHours - currentHours;
+        const durationMinutes = targetMinutes - currentMinutes;
+        const duration = durationHours * 60 * 60 * 1000 + durationMinutes * 60 * 1000
+        return duration;
+      }
+
+    render() {
+   
+    return (
+      <form onSubmit={ (e) => {
+        e.preventDefault(); 
+        this.props.smsToggle(); 
+        const duration = this.duration(e,this.state.time);
```

Comment:
its more commonly seen to extract methods that are more involved like the onSubmit you have here outside of the render method and into its own method. makes for easier readability in the jsx and a few other things, look into no side effects inside the render method